server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -627,5 +641,11 @@ private void deleteOriginFiles() throws IOException {
         logger.warn("clean origin file failed, origin file is : {}", originFilePath);
       }
     }
+
+    private ByteBuffer expandBufferAndUpdateMemoryTracker(int oldCapacity, int newCapacity) {
+      memoryTracker.releaseSortMemory(oldCapacity);
+      memoryTracker.reserveSortMemory(newCapacity);
+      return ByteBuffer.allocateDirect(newCapacity);
```

Comment:
If we need to call memoryTracker.sortMemoryReady to check here, how about adding a lock for `expandBufferAndUpdateMemoryTracker` instead of `sortMemoryReady`?

Reply:
sortMemoryReady is read-only, I think it's safe not to add lock anywhere.